sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala

```diff
@@ -1006,338 +417,12 @@ class HiveThriftCleanUpScratchDirSuite extends HiveThriftJdbcTest{
-
-  private var listeningPort: Int = _
-  protected def serverPort: Int = listeningPort
-
-  protected val hiveConfList = "a=avalue;b=bvalue"
-  protected val hiveVarList = "c=cvalue;d=dvalue"
-  protected def user = System.getProperty("user.name")
-
-  protected var warehousePath: File = _
-  protected var metastorePath: File = _
-  protected def metastoreJdbcUri = s"jdbc:derby:;databaseName=$metastorePath;create=true"
-
-  private val pidDir: File = Utils.createTempDir(namePrefix = "thriftserver-pid")
-  protected var logPath: File = _
-  protected var operationLogPath: File = _
-  protected var lScratchDir: File = _
-  private var logTailingProcess: Process = _
-  private var diagnosisBuffer: ArrayBuffer[String] = ArrayBuffer.empty[String]
-
-  protected def extraConf: Seq[String] = Nil
-
-  protected def serverStartCommand(port: Int) = {
-    val portConf = if (mode == ServerMode.binary) {
-      ConfVars.HIVE_SERVER2_THRIFT_PORT
-    } else {
-      ConfVars.HIVE_SERVER2_THRIFT_HTTP_PORT
-    }
-
-    val driverClassPath = {
-      // Writes a temporary log4j.properties and prepend it to driver classpath, so that it
-      // overrides all other potential log4j configurations contained in other dependency jar files.
-      val tempLog4jConf = Utils.createTempDir().getCanonicalPath
-
-      Files.write(
-        """log4j.rootCategory=DEBUG, console
-          |log4j.appender.console=org.apache.log4j.ConsoleAppender
-          |log4j.appender.console.target=System.err
-          |log4j.appender.console.layout=org.apache.log4j.PatternLayout
-          |log4j.appender.console.layout.ConversionPattern=%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n
-        """.stripMargin,
-        new File(s"$tempLog4jConf/log4j.properties"),
-        StandardCharsets.UTF_8)
-
-      tempLog4jConf
-    }
-
-    s"""$startScript
-       |  --master local
-       |  --hiveconf ${ConfVars.METASTORECONNECTURLKEY}=$metastoreJdbcUri
-       |  --hiveconf ${ConfVars.METASTOREWAREHOUSE}=$warehousePath
-       |  --hiveconf ${ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST}=localhost
-       |  --hiveconf ${ConfVars.HIVE_SERVER2_TRANSPORT_MODE}=$mode
-       |  --hiveconf ${ConfVars.HIVE_SERVER2_LOGGING_OPERATION_LOG_LOCATION}=$operationLogPath
-       |  --hiveconf ${ConfVars.LOCALSCRATCHDIR}=$lScratchDir
-       |  --hiveconf $portConf=$port
-       |  --driver-class-path $driverClassPath
-       |  --driver-java-options -Dlog4j.debug
-       |  --conf spark.ui.enabled=false
-       |  ${extraConf.mkString("\n")}
-     """.stripMargin.split("\\s+").toSeq
-  }
-
-  /**
-   * String to scan for when looking for the thrift binary endpoint running.
-   * This can change across Hive versions.
-   */
-  val THRIFT_BINARY_SERVICE_LIVE = "Starting ThriftBinaryCLIService on port"
-
-  /**
-   * String to scan for when looking for the thrift HTTP endpoint running.
-   * This can change across Hive versions.
-   */
-  val THRIFT_HTTP_SERVICE_LIVE = "Started ThriftHttpCLIService in http"
```

Comment:
If this becomes flaky because of the port, you could set port 0, and turn the use of these lines into a regex that would parse the port that got assigned...